### ISSUE TITLE
⚠Critical Cross-Site Injection (XSS) in TextArea elements⚠

### ISSUE THREAD

**Issue by Yato03** [drorganvidez/evidentia#156]
# [Critical Cross-Site Injection (XSS)](https://owasp.org/www-community/attacks/xss/) in TextArea elements

## Vulnerability description
**HTML injection** and **JavaScript execution** in forms related to TextArea funcionality.

![textArea](https://github.com/user-attachments/assets/07d7c22f-b4ba-4aee-95a6-c9156801bbe0)

This vulnerability allows attackers to inject malicious code to **impersonate a victim user** by looking to an evidence or an user description. If the victim browser is outdated, the vulnerability allows to **get remote access control** of the victim machine.

## To Reproduce
This is an example of malicious use of XSS. In this example I'll use two users: John Doe, a normal user and Margaret Hendricks, a coordinator. John Doe is going to make an evil evidence in order to accept itself when Margaret views it. 

1. **Creation of the evil evidence**. John create an evidence with a JavaScript payload:

![1](https://github.com/user-attachments/assets/0c01e523-0290-47ce-9c76-fce4b3b8b754)

We save it in draft in order to save the payload in the database. By now, the payload doesn't work.

**Payload**:
```html
<script>var l = window.location.href.split('/'); var index = l.length-1; window.location="http://localhost/21/coordinator/evidence/accept/"+l[index];</script> 
````
> Change the domain name by yours.

2. John edit again the evidence:

![2](https://github.com/user-attachments/assets/c6e4cad8-d3bd-4780-91e2-d8d034fe2c05)

3. John save the evidence in draft again.

![2-5](https://github.com/user-attachments/assets/f4205999-7624-443b-b6ca-2cf933930bfe)

By now, the payload is working. John's ittention is saving the payload working in the database.

4. John publish the evidence.

![3](https://github.com/user-attachments/assets/bc8eaf4b-d44d-4a43-be44-c1cd345642b8)

Now, John publish it to send it to Margaret.

5. Margaret views the evidence

![4](https://github.com/user-attachments/assets/431e7046-9691-4969-9361-98fe3f252b71)

When opened, the malicious code is executed.

6. The payload successfully accept the evidence.

![5](https://github.com/user-attachments/assets/425f537b-40aa-4ccc-ad50-983ba90e434b)

**With a more complex payload we can achieve exploits like changing the password of the victim.**

## Why is it vulnerable?

When we send the payload the first time, it isn't interpreted:

![sending_payload](https://github.com/user-attachments/assets/a75b712a-6194-42ca-883d-c81c3069a71e)

When we open it, the actual functionality of interpreting the styles like bold, italic, etc. render the malicious HTML tags.

![image](https://github.com/user-attachments/assets/1706e1b6-187d-4e4f-8537-4e639a453de7)

Now, if we save it again, the payload working is uploaded to the database and ready for being used.

To sum up, I think problem is how the system can save it in a safe way and not letting the textarea to render it.
